muzzle/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/ReferencesPrinter.java

```diff
@@ -26,7 +26,7 @@
    */
   public static void printMuzzleReferences() {
     for (InstrumentationModule instrumentationModule :
-        ServiceLoader.load(InstrumentationModule.class)) {
+        ServiceLoader.load(InstrumentationModule.class, ReferencesPrinter.class.getClassLoader())) {
```

Comment:
Why does this differ from the current context classloader?

Reply:
That's what surprised me too a bit - apparently the current context CL is still the gradle CL; while we want to use the instrumentation CL for everything that happens in this class.

Reply:
Should we do similar to https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/92a69c33097c7fc0d3d2453205c2f44c1289fbda/gradle-plugins/src/main/kotlin/io.opentelemetry.instrumentation.muzzle-check.gradle.kts#L219 ?

Reply:
Done.